packages/runtime/src/runtime/modal.tsx

```diff
@@ -89,7 +98,12 @@ export const ModalWrapper: React.FC<PropsWithChildren> = ({ children }) => {
       {
         visible: true,
         content: newContent,
-        options: newOptions,
+        options: evaluate(
+          screen as ScreenContextDefinition,
+          (isString(newOptions) ? newOptions : JSON.stringify(newOptions))
+            // eslint-disable-next-line prefer-named-capture-group
+            ?.replace(/['"]\$\{([^}]*)\}['"]/g, "$1"),
+        ),
```

Comment:
Instead of using `evaluate` directly use `useEvaluate` like the `useNavigateScreen` hook. Requires less dependencies and is more performant.